File: backend/app/schemas/group.py
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime
from uuid import UUID


class GroupBase(BaseModel):
    name: str = Field(..., description="Name of the group")
    description: Optional[str] = Field(None, description="Description of the group")


class GroupCreate(GroupBase):
    pass


class GroupUpdate(GroupBase):
    name: Optional[str] = None
    description: Optional[str] = None


class GroupResponse(GroupBase):
    id: UUID
    created_by_id: UUID
    created_at: datetime
    updated_at: datetime
    deleted_at: Optional[datetime] = None
    created_at_epoch: int

    class Config:
        from_attributes = True 

File: frontend/src/types/group.ts
import { type UUID } from "./endpoint"

export interface Group {
  id: UUID
  name: string
  description: string | null
  created_at: string
  updated_at: string
  created_by_id: UUID
  deleted_at: string | null
  created_at_epoch: number
}

export interface CreateGroupInput {
  name: string
  description: string
}

export interface UpdateGroupInput {
  name?: string
  description?: string
} 